### PR TITLE
Fix: the problem with duplicate filename of iOS

### PR DESCRIPTION
**Pull request by CaiJingLong**

### CODE REVIEW

**Comment by CaiJingLong** [comment]
The new file name contains id of PHAsset.

image

File: ios/Classes/core/PMManager.m [hunk 1]
```diff
@@ -479,6 +479,12 @@ - (void)getFullSizeFileWithId:(NSString *)id isOrigin:(BOOL)isOrigin subtype:(in
                 return;
             }
         }
+        if (@available(macOS 14.0, *)) {
+            if (asset.isLivePhoto && (subtype & PHAssetMediaSubtypePhotoLive) == PHAssetMediaSubtypePhotoLive) {
+                [self fetchLivePhotosFile:asset handler:handler progressHandler:progressHandler];
+                return;
+            }
+        }
```
Comment (CaiJingLong) [inline ios/Classes/core/PMManager.m:487]: When I was looking at the problem with the file name, I found that macOS and iOS have different logic about getting livePhoto, so I add the code.

**Comment by CaiJingLong** [comment]
In Android, the access permission itself depends on the Activity, so the code contains the code for dynamic permission checking, which requires your FlutterView to be Attached to an Activity, and it is performed according to the standard Android specification (that is, the channel called by each dart methods are wrapped in code for dynamic permission checking).

And your background service does not seem to depend on Activity, so you need to use ignorePermissionCheck to ignore PhotoManager's permission check on the premise of ensuring that you have permission.

```dart
// Call the method to ignore permission check
await PhotoManager.setIgnorePermissionCheck(true);
```

File: ios/Classes/core/PMManager.m [hunk 2]
```diff
@@ -714,7 +721,7 @@ - (NSString *)makeAssetOutputPath:(PHAsset *)asset
     } else {
         filename = [asset valueForKey:@"filename"];
     }
-    filename = [NSString stringWithFormat:@"%@%@_%@", modifiedDate, isOrigin ? @"_o" : @"", filename];
+    filename = [NSString stringWithFormat:@"%@_%@%@_%@", id, modifiedDate, isOrigin ? @"_o" : @"", filename];
```
Comment (AlexV525) [inline ios/Classes/core/PMManager.m:724]: Would it be too long if we added the ID here?

Comment (CaiJingLong) [reply ios/Classes/core/PMManager.m:724]: Other suggestion?

The id of asset will add 44 chars.
The biggest of darwin filename is 255.

Comment (AlexV525) [reply ios/Classes/core/PMManager.m:724]: Maybe compress the ID into 6~8 characters ID?

Comment (CaiJingLong) [reply ios/Classes/core/PMManager.m:724]: I'm not sure what is the probability that a certain substring of the id will be repeated.

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**